cluster/gce/windows/k8s-node-setup.psm1

```diff
@@ -1529,349 +1529,416 @@ function Start_Containerd {
-  # For 64 bits app, the registry path is 'HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall'
-  # for 32 bits app, it's 'HKLM:\SOFTWARE\Wow6432Node\Microsoft\Windows\CurrentVersion\Uninstall'
-  # StackdriverLogging is installed as 32 bits app
-  $x32_app_reg = 'HKLM:\SOFTWARE\Wow6432Node\Microsoft\Windows\CurrentVersion\Uninstall'
-  $uninstall_string = (Get-ChildItem $x32_app_reg | Get-ItemProperty | Where-Object {$_.DisplayName -match "Stackdriver"}).UninstallString
-  if (-not [string]::IsNullOrEmpty($uninstall_string)) {
-    try {
-      Start-Process -FilePath "$uninstall_string" -ArgumentList "/S" -Wait
-    } catch {
-      Log-Output "Exception happens during uninstall logging agent, so remove the registry key at least"
-      Remove-Item -Path "$x32_app_reg\GoogleStackdriverLoggingAgent\"
-    }
-  }
-
-  #  If we chose reboot after uninstallation, the root folder would be clean.
-  #  But since we couldn't reboot, so some files & folders would be left there,
-  #  which could block the re-installation later on, so clean it up
-  if(Test-Path $STACKDRIVER_ROOT){
-    Remove-Item -Force -Recurse $STACKDRIVER_ROOT
-  }
+  $logging_status = (Get-Service $LOGGINGAGENT_SERVICE -ErrorAction Ignore).Status
+  return -not [string]::IsNullOrEmpty($logging_status)
 }
 
-# Installs the Stackdriver logging agent according to
-# https://cloud.google.com/logging/docs/agent/installation.
-# TODO(yujuhong): Update to a newer Stackdriver agent once it is released to
-# support kubernetes metadata properly. The current version does not recognizes
-# the local resource key "logging.googleapis.com/local_resource_id", and fails
-# to label namespace, pod and container names on the logs.
+# Installs the logging agent according to https://docs.fluentbit.io/manual/installation/windows#
+# Also installs fluent bit stackdriver exporter
 function Install-LoggingAgent {
-  # Remove the existing storage.json file if it exists. This is a workaround
-  # for the bug where the logging agent cannot start up if the file is
-  # corrupted.
-  Remove-Item `
-      -Force `
-      -ErrorAction Ignore `
-      ("$STACKDRIVER_ROOT\LoggingAgent\Main\pos\winevtlog.pos\worker0\" +
-       "storage.json")
-
   if (IsLoggingAgentInstalled) {
-    # Note: we should reinstall the Stackdriver agent if $REDO_STEPS is true
+    # Note: we should reinstall the agent if $REDO_STEPS is true
     # here, but we don't know how to run the installer without it prompting
-    # when Stackdriver is already installed. We dumped the strings in the
+    # when logging agent is already installed. We dumped the strings in the
     # installer binary and searched for flags to do this but found nothing. Oh
     # well.
-    Log-Output ("Skip: Stackdriver logging agent is already installed")
+    Log-Output ("Skip: Fluentbit logging agent is already installed")
     return
   }
+  
+  DownloadAndInstall-LoggingAgentService
+  Remove-Item -Force -Recurse $LOGGINGAGENT_INSTALLER_DIR
```

Comment:
Sorry if my previous comment wasn't clear. Can you try something like this?: https://gist.github.com/pjh/604c34b342d0560f6b20abce7c28e08b. That code is consistent with the approach we use a few times elsewhere in this file.

Having to make `LOGGINGAGENT_INSTALLER_DIR` a constant (having one function clean up another function's temporary dir) is a sign that something can probably be made simpler.

Reply:
done. please review